tests/integration/common-operator-integ-suite.sh

```diff
@@ -14,118 +14,204 @@
+    COMMAND="oc"
+  fi
 
-BRANCH="${BRANCH:-maistra-3.0}"
+  echo "Using command: ${COMMAND}"
 
-if [ "${OCP}" == "true" ]; then
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
-else
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
-fi
+  if [ "${OCP}" == "true" ]; then
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
+  else
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
+  fi
 
-TIMEOUT="3m"
+  TIMEOUT="3m"
+}
 
-check_ready() {
-    local NS=$1
-    local POD_NAME=$2
-    local DEPLOYMENT_NAME=$3
+get_internal_registry() {
+  # Validate that the internal registry is running, configure the variable to be used in the Makefile. 
+  # If there is no internal registry, the test can't be executed targeting to the internal registry
+
+  # Check if the registry pods are running
+  ${COMMAND} get pods -n openshift-image-registry --no-headers | grep -v "Running" && echo "It looks like the OCP image registry is not deployed or Running. This tests scenario requires it. Aborting." && exit 1
 
-    echo "Check POD: NAMESPACE: \"${NS}\"   POD NAME: \"${POD_NAME}\""
+  # Check if default route already exist
+  if [ -z "$(${COMMAND} get route default-route -n openshift-image-registry -o name)" ]; then
+    echo "Route default-route does not exist, so you can perform patching here."
+    ${COMMAND} patch configs.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
+  
     timeout --foreground -v -s SIGHUP -k ${TIMEOUT} ${TIMEOUT} bash --verbose -c \
-    "until $COMMAND  get pod --field-selector=status.phase=Running -n ${NS} | grep ${POD_NAME}; do sleep 5; done"
+      "until ${COMMAND} get route default-route -n openshift-image-registry &> /dev/null; do sleep 5; done && echo 'The 'default-route' has been created.'"
+  fi
+
+  # Get the registry route
+  URL=$(${COMMAND} get route default-route -n openshift-image-registry --template='{{ .spec.host }}')
+  # Hub will be equal to the route url/project-name(NameSpace) 
+  export HUB="${URL}/${NAMESPACE}"
+  echo "Internal registry URL: ${HUB}"
+
+  # Create namespace where operator will be located
+  # This is needed because the roles are created in the namespace where the operator is deployed
+  ${COMMAND}  create namespace "${NAMESPACE}" || true
+
+  # Adding roles to avoid the need to be authenticated to push images to the internal registry
+  # Using envsubst to replace the variable NAMESPACE in the yaml file
+  envsubst < "${WD}/config/role-bindings.yaml" | ${COMMAND} apply -f -
+
+  # Login to the internal registry when running on CRC
+  # Take into count that you will need to add before the registry URL as Insecure registry in "/etc/docker/daemon.json"
+  if [[ ${URL} == *".apps-crc.testing"* ]]; then
+    echo "Executing Docker login to the internal registry"
+    docker login -u $(${COMMAND} whoami) -p $(${COMMAND} whoami -t) ${URL}
```

Comment:
```suggestion
    ${COMMAND} whoami -t | docker login -u "$(${COMMAND} whoami)" --password-stdin ${URL}
```

Reply:
Also, since this entire function is only used with OpenShift, does it really make sense to use `${COMMAND}` instead of simply `oc`? Why make the code harder to read?

Reply:
Was only intended to have everything with the same format. But we can change it to make easy to read

Reply:
Change done and added a message for when the docker login login fails to aware the user that is needed to be added insecure registry when you are running the test locally and not from the prow job